### ISSUE TITLE
Is it possible to individualize vim tabs and splits?

### ISSUE THREAD

**Issue by bert88sta** [extr0py/oni#545]
I feel as though putting each tab in its own container, which then in turn contains multiple splits, could be beneficial for rendering. If we're forced to track where the edits are taking place or where the scrolling is, then we don't have to redraw other splits.

Also, one of the things I've always wanted (that seems impossible in vim) are split borders. I've abandoned airline in favor of the builtin colorscheme and Oni's statusline, but this annoys me:

<img width="1680" alt="screen shot 2017-07-20 at 9 45 09 am" src="https://user-images.githubusercontent.com/8421641/28420440-28cb0f30-6d30-11e7-829b-9f09bfb081cc.png">

(these are 2 separate splits, but if they were the same language it would be hard to tell where one begins and one ends)

**Comment by extr0py** [comment]
Yes, great suggestion. I'm hitting this too - sometimes it's difficult to know where the split boundaries are in the current implementation!

I'd like to provide more control over the actual split window UI - borders, drop shadow, etc. And explicitly controlling the rendering here will allow us to do fancier things and more in-depth performance optimizations in the buffer rendering and responsive. Some of the thinking around this is captured in #362.

**Comment by bryphe** [comment]
For this, I'm thinking about adding an `editor.splitMode` configuration, with a value of `oni` or `native` - having a value of `oni` would result in using containerized splits as you describe (along with multiplexed Neovim processes).

**Comment by badosu** [comment]
@bryphe Seems like this was addressed now that you can have multiple Oni windows, right?
